File: DevExpress-Demo/VB/LayoutControlDemo.Wpf/Data/SupportedDataTypesData_FluentAPI.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports System.ComponentModel.DataAnnotations
Imports System.Windows

Namespace LayoutControlDemo

    <MetadataType(GetType(SupportedDataTypesMetadata))>
    Public Class SupportedDataTypesData_FluentAPI

        Public Property IntProperty As Integer

        Public Property NullableIntProperty As Integer?

        Public Property DoubleProperty As Double

        Public Property NullableDoubleProperty As Double?

        Public Property BoolProperty As Boolean

        Public Property NullableBoolProperty As Boolean?

        Public Property CharProperty As Char

        Public Property NullableCharProperty As Char?

        Public Property EnumProperty As Gender

        Public Property NullableEnumProperty As Gender?

        Public Property StringProperty As String

        Public Property DateTimeProperty As Date

        Public Property NullableDateTimeProperty As Date?

        Public Property DecimalProperty As Decimal

        Public Property NullableDecimalProperty As Decimal?

        Public Property ComplexTypeProperty As Point

        Public Property CurrencyProperty As Decimal

        Public Property MultilineTextProperty As String

        Public Property PasswordProperty As String

        Public Property PhoneNumberProperty As String
    End Class

    Public Module SupportedDataTypesMetadata

        Public Sub BuildMetadata(ByVal builder As MetadataBuilder(Of SupportedDataTypesData_FluentAPI))
            builder.[Property](Function(x) x.CurrencyProperty).CurrencyDataType()
            builder.[Property](Function(x) x.MultilineTextProperty).MultilineTextDataType()
            builder.[Property](Function(x) x.PhoneNumberProperty).PhoneNumberDataType()
            builder.[Property](Function(x) x.PasswordProperty).PasswordDataType()
        End Sub
    End Module
End Namespace

File: DevExpress-Demo/CS/LayoutControlDemo.Wpf/Data/SupportedDataTypesData_FluentAPI.cs
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.ComponentModel.DataAnnotations;
using System.Windows;

namespace LayoutControlDemo {
    [MetadataType(typeof(SupportedDataTypesMetadata))]
    public class SupportedDataTypesData_FluentAPI {
        public int IntProperty { get; set; }
        public int? NullableIntProperty { get; set; }
        public double DoubleProperty { get; set; }
        public double? NullableDoubleProperty { get; set; }
        public bool BoolProperty { get; set; }
        public bool? NullableBoolProperty { get; set; }
        public char CharProperty { get; set; }
        public char? NullableCharProperty { get; set; }
        public Gender EnumProperty { get; set; }
        public Gender? NullableEnumProperty { get; set; }
        public string StringProperty { get; set; }
        public DateTime DateTimeProperty { get; set; }
        public DateTime? NullableDateTimeProperty { get; set; }
        public decimal DecimalProperty { get; set; }
        public decimal? NullableDecimalProperty { get; set; }
        public Point ComplexTypeProperty { get; set; }
        public decimal CurrencyProperty { get; set; }

        public string MultilineTextProperty { get; set; }
        public string PasswordProperty { get; set; }
        public string PhoneNumberProperty { get; set; }
    }
    public static class SupportedDataTypesMetadata  {
        public static void BuildMetadata(MetadataBuilder<SupportedDataTypesData_FluentAPI> builder) {
            builder.Property(x => x.CurrencyProperty)
                .CurrencyDataType();
            builder.Property(x => x.MultilineTextProperty)
                .MultilineTextDataType();
            builder.Property(x => x.PhoneNumberProperty)
                .PhoneNumberDataType();
            builder.Property(x => x.PasswordProperty)
                .PasswordDataType();
        }
    }
}
